backend/src/router.go

```diff
@@ -349,3 +351,48 @@ func getProfile(c *gin.Context) {
 		"user": users[0],
 	})
 }
+
+
+//not all of this is needed to make an update, at least according to the guide. We shall see.
+type ProfileUpdateData struct { 
+	ID        uint   `json:"-"`
+	FirstName string `json:"firstname"`
+	LastName  string `json:"lastname"`
+	Email     string `json:"email"`
+	Phone     string `json:"phone"`
+}
+
+// ADD DESCRIPTION
+func patchProfile(c *gin.Context) {
+	
+	token, err := c.Cookie("jwt")
+	if err != nil {
+		c.JSON(401, gin.H{
+			"error": "Requires an authenticated user.",
+		})
+		return
+	}
+	
+	claims, err := ParseJWT(token)
+	if err != nil {
+		c.JSON(500, gin.H{
+			"error": "Unable to parse JWT: " + err.Error() + ".",
+		})
+		return
+	}
+	
+	var users []User
+	//add error checking for nonexisting user 
+	//Note for self: https://blog.logrocket.com/how-to-build-a-rest-api-with-golang-using-gin-and-gorm/
+	
+	var edits ProfileUpdateData
+	if err := c.ShouldBindJSON(&edits); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
```

Comment:
Minor nitpick - for consistency, let's stick to directly using `400` as the status rather than the constant. We can open an issue to make this change to all of the routes later; didn't realize this was a thing.

Reply:
Fixed to a 400 and added the documentation. I had planned on adding it soon, just wanted to get the init stuff working before I made docs and then would need to make major changes to it.